cvat/apps/engine/rq.py

```diff
@@ -0,0 +1,527 @@
+# Copyright (C) CVAT.ai Corporation
+#
+# SPDX-License-Identifier: MIT
+
+from __future__ import annotations
+
+from abc import ABCMeta, abstractmethod
+from datetime import datetime
+from typing import Any, Callable, Optional, Protocol, Union
+from uuid import UUID
+
+import attrs
+from django.conf import settings
+from django.db.models import Model
+from django.utils import timezone
+from django_rq.queues import DjangoRQ
+from rq.job import Dependency as RQDependency
+from rq.job import Job as RQJob
+from rq.registry import BaseRegistry as RQBaseRegistry
+
+from cvat.apps.engine.types import ExtendedRequest
+
+from .models import RequestAction, RequestSubresource, RequestTarget
+
+
+class RQJobMetaField:
+    class UserField:
+        ID = "id"
+        USERNAME = "username"
+        EMAIL = "email"
+
+    class RequestField:
+        UUID = "uuid"
+        TIMESTAMP = "timestamp"
+
+    # common fields
+    FORMATTED_EXCEPTION = "formatted_exception"
+    REQUEST = "request"
+    USER = "user"
+    PROJECT_ID = "project_id"
+    TASK_ID = "task_id"
+    JOB_ID = "job_id"
+    LAMBDA = "lambda"
+    ORG_ID = "org_id"
+    ORG_SLUG = "org_slug"
+    STATUS = "status"
+    PROGRESS = "progress"
+    TASK_PROGRESS = "task_progress"
+    # export specific fields
+    RESULT_URL = "result_url"
+    RESULT = "result"
+    FUNCTION_ID = "function_id"
+    EXCEPTION_TYPE = "exc_type"
+    EXCEPTION_ARGS = "exc_args"
+    TMP_FILE = "tmp_file"
+
+
+class WithMeta(Protocol):
+    meta: dict[str, Any]
+
+
+class _AbstractRQMetaAttribute:
+    def __init__(
+        self, key: str, *, optional: bool = False, validator: Callable | None = None
+    ) -> None:
+        assert validator is None or callable(validator), "validator must be callable"
+        self._key = key
+        self._validator = validator
```

Comment:
The `validator` parameter ought to be moved to `_SettableRQMetaAttribute`, since it doesn't do anything in an immutable attribute.